zk/datastream/server/datastream_populate.go

```diff
@@ -125,20 +125,22 @@ func WriteBlocksToStream(
 			return err
 		}
 
-		gersInBetween, err := reader.GetBatchGlobalExitRootsProto(prevBatchNum, batchNum)
+		nextBatchNum, err := reader.GetBatchNoByL2Block(currentBlockNumber + 1)
 		if err != nil {
 			return err
 		}
 
-		l1InfoMinTimestamps := make(map[uint64]uint64)
+		// a 0 next batch num here would mean we don't know about the next batch so must be at the end of the batch
+		isBatchEnd := nextBatchNum == 0 || nextBatchNum > batchNum
 
-		// as we're writing to the stream here and not just generating for the executor we want to run in standard mode
-		mode := StandardOperationMode
+		gersInBetween, err := reader.GetBatchGlobalExitRootsProto(prevBatchNum, batchNum)
+		if err != nil {
+			return err
+		}
 
-		// we do not want to force the closing of a batch when writing to the stream - this is only the domain of talking to the executor
-		const forceBatchEnd = false
+		l1InfoMinTimestamps := make(map[uint64]uint64)
```

Comment:
Just have this `make` directly in the call, and also add a comment that this param will be removed in the future version and it is unused.

Reply:
I can make it directly in the call, but this is still very much used.  We use it to pass the min timestamps to the executor, without it the calls to it will fail when changes to the l1 info root are present in the batch

Reply:
Ah, that is for the datastream, not for the executor code, sorry then.